### PR TITLE
Update getProcessorName routines

### PR DESCRIPTION
**Pull request by dchopra001**
This commit cleans up and simplifies usage of the getProcessorName routines in z/codegen's implementation of OMRCPU.

Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

**Comment by dchopra001** [comment]
@fjeremic Can I get a review?

File: compiler/z/codegen/OMRInstruction.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ OMR::Z::Instruction::Instruction(TR::CodeGenerator* cg, TR::InstOpCode::Mnemonic
    OMR::Instruction(cg, op, node),
    CTOR_INITIALIZER_LIST
    {
-   TR_ASSERT_FATAL(cg->comp()->target().cpu.isAtLeast(_opcode.getMinimumALS()), "Processor detected does not support instruction %s\n", _opcode.getMnemonicName());
+   TR_ASSERT_FATAL(cg->comp()->target().cpu.isAtLeast(_opcode.getMinimumALS()), "Processor detected(%s) does not support instruction %s\n", cg->comp()->target().cpu.getProcessorName(), _opcode.getMnemonicName());
```
Comment (fjeremic) [inline compiler/z/codegen/OMRInstruction.cpp:89]:
```suggestion
   TR_ASSERT_FATAL(cg->comp()->target().cpu.isAtLeast(_opcode.getMinimumALS()), "Processor detected (%s) does not support instruction %s\n", cg->comp()->target().cpu.getProcessorName(), _opcode.getMnemonicName());
```

Similarly below.

File: compiler/z/env/OMRCPU.cpp [hunk 2]
```diff
@@ -197,58 +197,11 @@ OMR::Z::CPU::supportsFeatureOldAPI(uint32_t feature)
...
-         TR_ASSERT(false, "Invalid Archiecture Type: %d", arch);
-         result = "Invalid Architecture type!";
-         break;
-      }
-   return result;
-   }
-
-const char*
-OMR::Z::CPU::getProcessorName(OMRProcessorArchitecture arch)
+OMR::Z::CPU::getProcessorName()
```
Comment (fjeremic) [inline compiler/z/env/OMRCPU.cpp:200]: I think we need a bit more work here. We now have duplication between OMR/OpenJ9. See:

https://github.com/eclipse-openj9/openj9/blob/a82dc74872dc55d92d90b7af4751bdfa0980b0f9/runtime/compiler/env/ProcessorDetection.cpp#L294-L542

We need to eliminate this duplication while we're in the area here. Do you want to tackle this here or open another issue to remove this? IMO `getProcessorName` being reviewed here should be in the common CPU class which just returns "Unknown" and then we override on each platform via extensible classes like you do here.

Comment (dchopra001) [reply compiler/z/env/OMRCPU.cpp:200]: > IMO getProcessorName being reviewed here should be in the common CPU class which just returns "Unknown" and then we override on each platform via extensible classes like you do here

Can't we do all this work for all the CPUs in the common CPU class? If so then isn't it better to do it there?

File: compiler/z/codegen/OMRInstruction.cpp (hunk 1 above)
Comment (dchopra001) [reply compiler/z/codegen/OMRInstruction.cpp]: Fixed in 03905d340e70e72dd1c91acc45dd06949c47c9dc

File: compiler/z/env/OMRCPU.cpp (hunk 2 above)
Comment (fjeremic) [reply compiler/z/env/OMRCPU.cpp:200]: If you do it at the common level you'll have to have a bunch of `if (isZ)`, `else if (isPower)`, etc. which is not really what we want.

Comment (dchopra001) [reply compiler/z/env/OMRCPU.cpp:200]: Changes made in 03905d340e70e72dd1c91acc45dd06949c47c9dc. This PR is ready for another review.

**Review by fjeremic — APPROVED** [review]

**Comment by fjeremic** [comment]
Jenkins build all

**Comment by fjeremic** [comment]
@dchopra001 see AArch64 failure.

**Comment by dchopra001** [comment]
@fjeremic I think it should be fixed now.

**Comment by fjeremic** [comment]
Jenkins build all

**Review by fjeremic — APPROVED** [review]

**Pull request merged**